lib/migration/safe_migrate.rb

```diff
@@ -78,6 +78,20 @@ def self.enable!
     return if ENV["RAILS_ENV"] == "production"
 
     @@migration_sqls = []
+    @@activerecord_remove_indexes = []
+
+    ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.class_eval do
+      alias_method :original_remove_index, :remove_index
+
+      def remove_index(table_name, column_name = nil, **options)
+        @@activerecord_remove_indexes << (
+          options[:name] || index_name(table_name, options.merge(column: column_name))
+        )
+
+        # Call the original method
+        original_remove_index(table_name, column_name, **options)
+      end
+    end
```

Comment:
Monkey patching ActiveRecord here because `remove_index` with the `if_exists: true` option does not execute a `DROP INDEX ...` statement if the index exists.